### PR TITLE
Remove discussions of MBEDTLS_USE_PSA_CRYPTO in API documentation

### LINKED ISSUE #9632: Remove discussions of MBEDTLS_USE_PSA_CRYPTO in documentation
Remove discussions of `MBEDTLS_USE_PSA_CRYPTO` in documentation. The documentation should now only what was formerly the behavior when `MBEDTLS_USE_PSA_CRYPTO` is enabled.

Note that while `docs/use-psa-crypto.md` will be partly obsolete (e.g. “why enable” is no longer relevant), some of the information is still relevant (at least limitations).

Follow-up to https://github.com/Mbed-TLS/mbedtls/issues/9630

### PR DESCRIPTION
**Pull request by yanesca**
## Description

Remove discussions of MBEDTLS_USE_PSA_CRYPTO in API documentation. Resolves partially #9632.

## PR checklist

Please remove the segment/s on either side of the | symbol as appropriate, and add any relevant link/s to the end of the line.
If the provided content is part of the present PR remove the # symbol.

- [x] **changelog** not required because: documentation only
- [x] **development PR** this 
- [x] **framework PR** not required
- [x] **3.6 PR** not required because: 4.0 only
- [x] **2.28 PR** not required because: 4.0 only
- **tests**  not required because: documentation only

### CODE REVIEW

File: include/mbedtls/mbedtls_config.h
Comment (gilles-peskine-arm) [reply include/mbedtls/mbedtls_config.h:853]: Bug report for 3.6: https://github.com/Mbed-TLS/mbedtls/issues/9790

Follow-up task for 4.0: https://github.com/Mbed-TLS/mbedtls/issues/9791

File: configs/ext/tfm_mbedcrypto_config_profile_medium.h
Comment (gilles-peskine-arm) [reply configs/ext/tfm_mbedcrypto_config_profile_medium.h:523]: As discussed in Slack, we do need to change this file, so please update the readme to say that this is currently not a verbatim copy, but a preview of how TFM should be adapted for 4.0.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
This mostly looks good, and I'm satisfied with completeness for the subtask “API documentation”. There are a few problems in `mbedtls_config.h`. There are conflicts with #9771, so it would be better to wait until that is merged for rework and re-review.

File: include/mbedtls/mbedtls_config.h
Comment (yanesca) [reply include/mbedtls/mbedtls_config.h:645]: I am sorry I am not sure I can see which of these limitations has gone. Could you please elaborate?

Comment (yanesca) [reply include/mbedtls/mbedtls_config.h:645]: Raised https://github.com/Mbed-TLS/mbedtls/issues/9817 for tracking.

Comment (gilles-peskine-arm) [reply include/mbedtls/mbedtls_config.h]: Sorry, I must have forgotten to actually write a sentence here, or I made a mistake when consolidating/separating comments. Grammatically, the current paragraph contains two limitations (one per sentence). The first limitation “… not using PSA” is not really meaningful anymore, since users must assume that everything is “using PSA” — users must call `psa_crypto_init` and must enable `PSA_WANT_xxx`. So I thought we should remove that sentence: it's now an implementation detail, no longer a statement about the interface. The sentence isn't misleading though, and it's informative in a non-semantic way, so it can stay.

File: include/mbedtls/mbedtls_config.h
Comment (yanesca) [reply include/mbedtls/mbedtls_config.h:643]: Using or not using PSA might be an implementation detail, but it has user visible consequences. (eg. using or not using drivers)

**Review by davidhorstmann-arm — APPROVED** [review]
Changes LGTM and there are no documentation mentions of USE_PSA in `include` or `tf-psa-crypto/include`, which are the public-facing parts.

**Pull request merged**
